app/assets/javascripts/discourse/components/pwa-install-banner.js.es6

```diff
@@ -0,0 +1,61 @@
+import {
+  default as computed,
+  on
+} from "ember-addons/ember-computed-decorators";
+
+const USER_DISMISSED_PROMPT_KEY = "dismissed-pwa-install-banner";
+
+export default Ember.Component.extend({
+  deferredInstallPromptEvent: null,
+
+  _handleInstallPromptEvent(event) {
+    // Prevent Chrome 76+ from automatically showing the prompt
+    event.preventDefault();
+    // Stash the event so it can be triggered later
+    this.set("deferredInstallPromptEvent", event);
+  },
+
+  @on("init")
+  _registerListener() {
+    this._promptEventHandler = Ember.run.bind(
+      this,
+      this._handleInstallPromptEvent
+    );
+    window.addEventListener("beforeinstallprompt", this._promptEventHandler);
+  },
+
+  @computed
+  bannerDismissed: {
+    set(value) {
+      localStorage.setItem(USER_DISMISSED_PROMPT_KEY, value);
```

Comment:
It would be better I think to use the `KeyValueStore` for this. It has some extra checks about if localStorage works.